substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -905,11 +905,157 @@ protected Object socketGetOption(int opt) throws SocketException {
+            //  1382          return;
+            /* Unreachable. */
+            //  1383      }
+        }
+
+        //  1385      /*
+        //  1386       * Setting the multicast interface handled separately
+        //  1387       */
+        //  1388      if (opt == java_net_SocketOptions_IP_MULTICAST_IF ||
+        //  1389          opt == java_net_SocketOptions_IP_MULTICAST_IF2) {
+        if (opt == SocketOptions.IP_MULTICAST_IF || opt == SocketOptions.IP_MULTICAST_IF2) {
+            //  1391          setMulticastInterface(env, this, fd, opt, value);
+            Util_java_net_PlainDatagramSocketImpl.setMulticastInterface(fd, opt, (InetAddress) value);
+            //  1392          return;
+            /* Unreachable. */
+            //  1393      }
+        }
+
+        //  1395      /*
+        //  1396       * Setting the multicast loopback mode handled separately
+        //  1397       */
+        //  1398      if (opt == java_net_SocketOptions_IP_MULTICAST_LOOP) {
+        if (opt == SocketOptions.IP_MULTICAST_LOOP) {
+            //  1399          setMulticastLoopbackMode(env, this, fd, opt, value);
+            Util_java_net_PlainDatagramSocketImpl.setMulticastLoopbackMode(fd, opt, value);
+            //  1400          return;
+            /* Unreachable. */
+            //  1401      }
+        }
+
+        //  1403      /*
+        //  1404       * Map the Java level socket option to the platform specific
+        //  1405       * level and option name.
+        //  1406       */
+        //  1407      if (NET_MapSocketOption(opt, &level, &optname)) {
+        if (JavaNetNetUtilMD.NET_MapSocketOption(opt, level_Pointer, optname_Pointer) != 0) {
+            //  1408          JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException", "Invalid option");
+            throw new SocketException("Invalid option");
+            //  1409          return;
+            /* Unreachable. */
+        }
+
+        //  1412      switch (opt) {
+        switch (opt) {
+            //  1413          case java_net_SocketOptions_SO_SNDBUF :
+            case SocketOptions.SO_SNDBUF:
+                //  1414          case java_net_SocketOptions_SO_RCVBUF :
+            case SocketOptions.SO_RCVBUF:
+                //  1415          case java_net_SocketOptions_IP_TOS :
+            case SocketOptions.IP_TOS: {
+                //  1417                  jclass cls;
+                //  1418                  jfieldID fid;
+                //  1419
+                //  1420                  cls = (*env)->FindClass(env, "java/lang/Integer");
+                //  1421                  CHECK_NULL(cls);
+                //  1422                  fid =  (*env)->GetFieldID(env, cls, "value", "I");
+                //  1423                  CHECK_NULL(fid);
+                //  1424
+                //  1425                  optval = (*env)->GetIntField(env, value, fid);
+                optval = (Integer) value;
```

Comment:
I find this confusing to read.  This is more like `optval = ((Integer) value).intValue()`.  `value` is an `Object`, which we cast to an `Integer`, from which we want the value.  Eclipse paints `value` red to indicate auto-unboxing.  I would rather have the access be explicit.

Hmm, and what is the failure if `value` is *not* an `Integer`?  It looks like we fail a guarantee in the HotSpot JVM.  I guess a `ClassCastException` is close enough.